src/components/CurrencyInputPanel/CurrencyInputPanel.component.tsx

```diff
@@ -74,6 +76,28 @@ export const CurrencyInputPanelComponent = ({
     }
   }, [disableCurrencySelect, isLoading])
 
+  const handleOnMax = useCallback(() => {
+    if (onMax && setIsMaxAmount) {
+      onMax()
+      setIsMaxAmount(true)
+    }
+  }, [onMax, setIsMaxAmount])
+
+  const handleOnCurrencySelect = useCallback(
+    (currency: Currency) => {
+      if (onCurrencySelect && currency) onCurrencySelect(currency, isMaxAmount)
+    },
+    [isMaxAmount, onCurrencySelect]
+  )
+
+  const handleOnUserInput = useCallback(
+    (value: string) => {
+      setIsMaxAmount(false)
+      if (onUserInput) onUserInput(value)
```

Comment:
Unnecessary check.
```js
//type of onUserInput
var onUserInput: (value: string) => void
```

Reply:
Addressed in [60ad9c7](https://github.com/levelkdev/swapr-dapp/pull/1013/commits/60ad9c7e9feff96ac053e0555e59897acb3af6fd). Thanks!